lib/config-generator.js

```diff
@@ -21,6 +21,17 @@ import { langQuestions, jsQuestions, mdQuestions, installationQuestions } from "
 // Helpers
 //-----------------------------------------------------------------------------
 
+/**
+ * Regular expression to check if a version requirement allows ESLint 9.22.0 or higher.
+ * Matches:
+ * - "^9" (any ^9.x.x version)
+ * - ">=9" (version 9.0.0 or higher)
+ * - "~9.22" or higher minor versions (9.22.x and above)
+ * - "9.22" or higher versions without prefix
+ * - Correctly handles full semver notation (e.g., 9.22.0, ~9.22.0)
+ */
+const ESLINT_9_PLUS_REGEX = /\^9|>=\s*9|(?:~|^)9\.(2[2-9]|[3-9]\d|\d{3,})/u;
```

Comment:
Maybe it would make sense to use the `semver` package instead? That way we could eliminate one source of bugs.

Reply:
updated 👍🏻